boards/feather_m4/src/lib.rs

```diff
@@ -1,103 +1,156 @@
-    pin d10 = a20,
-    /// Pin 11, PWM capable
-    pin d11 = a21,
-    /// Pin 12, PWM capable
-    pin d12 = a22,
-    /// Pin 13, which is also attached to
-    /// the red LED.  PWM capable.
-    pin d13 = a23,
-
-    /// The I2C data line
-    pin sda = a12,
-    /// The I2C clock line
-    pin scl = a13,
-
-    /// The SPI SCK
-    pin sck = a17,
-    /// The SPI MOSI
-    pin mosi = b23,
-    /// The SPI MISO
-    pin miso = b22,
-
-    /// The USB D- pad
-    pin usb_dm = a24,
-    /// The USB D+ pad
-    pin usb_dp = a25,
+hal::bsp_pins!(
+    PA02 {
+        /// Analog pin 0.  Can act as a true analog output
+        /// as it has a DAC (which is not currently supported
+        /// by this hal) as well as input.
+        name: a0,
+    }
+    PA05 {
+        /// Analog Pin 1
+        name: a1,
+    }
+    PB08 {
+        /// Analog Pin 2
+        name: a2,
+    }
+    PB09 {
+        /// Analog Pin 3
+        name: a3,
+    }
+    PA04 {
+        /// Analog Pin 4
+        name: a4,
+    }
+    PA06 {
+        /// Analog Pin 5
+        name: a5,
+    }
+    PB17 {
+        /// Pin 0, UART rx
+        name: d0,
+        aliases: {
+            AlternateC: UartRx
+        }
+    }
+    PB16 {
+        /// Pin 1, UART tx
+        name: d1,
+        aliases: {
+            AlternateC: UartTx
+        }
+    }
+    PA14 {
+        /// Pin 4, PWM capable
+        name: d4,
+    }
+    PA16 {
+        /// Pin 5, PWM capable
+        name: d5,
+    }
+    PA18 {
+        /// Pin 6, PWM capable
+        name: d6,
+    }
+    PB03 {
+        /// Neopixel Pin
+        name: neopixel,
+    }
+    PA19 {
+        /// Pin 9, PWM capable.  Also analog input (A7)
+        name: d9,
+    }
+    PA20 {
+        /// Pin 10, PWM capable
+        name: d10,
+    }
+    PA21 {
+        /// Pin 11, PWM capable
+        name: d11,
+    }
+    PA22 {
+        /// Pin 12, PWM capable
+        name: d12,
+    }
+    PA23 {
+        /// Pin 13, which is also attached to the red LED. PWM capable.
+        name: d13,
+        aliases: {
+            PushPullOutput: RedLed,
+            AlternateE: RedLedPwm
+        }
+    }
+    PA12 {
+        /// The I2C data line
+        name: sda,
+        aliases: {
+            AlternateC: Sda
+        }
+    }
+    PA13 {
+        /// The I2C clock line
+        name: scl,
+        aliases: {
+            AlternateC: Scl
+        }
+    }
+    PA17 {
+        /// The SPI SCK
+        name: sck,
+        aliases: {
+            AlternateC: Sck
+        }
+    }
+    PB23 {
+        /// The SPI MOSI
+        name: mosi,
+        aliases: {
+            AlternateC: Mosi
+        }
+    }
+    PB22 {
+        /// The SPI MISO
+        name: miso,
+        aliases: {
+            AlternateC: Miso
+        }
+    }
+    PA24 {
+        /// The USB D- pad
+        name: usb_dm,
+    }
+    PA25 {
+        /// The USB D+ pad
+        name: usb_dp,
+    }
 );
 
 /// Convenience for setting up the labelled SPI peripheral.
```

Comment:
@TDHolmes, thanks for working on this. I think the only thing I see missing is switching to `spi::v2` in `lib.rs`.